From c0c71e6c6d201e60bc91da0c2c888052e4819668 Mon Sep 17 00:00:00 2001
From: QtScript Qt 6 patch set <qtscript-qt6@local.invalid>
Date: Sun, 9 Aug 2026 18:42:32 +0200
Subject: [PATCH 2/2] Add QuickJS-compatible ScriptTools shell

---
 src/scripttools/debugging/qscriptenginedebugger.h |    2 --
 1 file changed, 2 deletions(-)

diff --git a/src/scripttools/debugging/qscriptenginedebugger.h b/src/scripttools/debugging/qscriptenginedebugger.h
index c45800b..771e344 100644
--- a/src/scripttools/debugging/qscriptenginedebugger.h
+++ b/src/scripttools/debugging/qscriptenginedebugger.h
@@ -122,8 +122,6 @@ Q_SIGNALS:
 private:
     Q_DECLARE_PRIVATE(QScriptEngineDebugger)
     Q_DISABLE_COPY(QScriptEngineDebugger)
-
-    Q_PRIVATE_SLOT(d_func(), void _q_showStandardWindow())
 };
 
 QT_END_NAMESPACE
