From b1dbffec126f4342cbbcf2e82ddd7a7b316e1990 Mon Sep 17 00:00:00 2001
From: Julien Maille <julien.maille@gmail.com>
Date: Sun, 9 Aug 2026 21:11:52 +0200
Subject: [PATCH 1/2] Advance QuickJS public compatibility slice

 tests/auto/qscriptvalue/tst_qscriptvalue.cpp |   2 -
 1 file changed, 2 deletions(-)

diff --git a/tests/auto/qscriptvalue/tst_qscriptvalue.cpp b/tests/auto/qscriptvalue/tst_qscriptvalue.cpp
index 0758803..abc24cb 100644
--- a/tests/auto/qscriptvalue/tst_qscriptvalue.cpp
+++ b/tests/auto/qscriptvalue/tst_qscriptvalue.cpp
@@ -1990,10 +1990,8 @@ void tst_QScriptValue::getSetProperty_gettersAndSettersThrowErrorJS()
     QScriptValue object = eng.evaluate("o");
     QVERIFY(!eng.hasUncaughtException());
     QScriptValue ret = object.property("foo");
-    QEXPECT_FAIL("", "QTBUG-17616: Exception thrown from js function are not returned by the JSC port", Continue);
     QVERIFY(ret.isError());
     QVERIFY(eng.hasUncaughtException());
-    QEXPECT_FAIL("", "QTBUG-17616: Exception thrown from js function are not returned by the JSC port", Continue);
     QVERIFY(ret.strictlyEquals(eng.uncaughtException()));
     QCOMPARE(eng.uncaughtException().toString(), QLatin1String("Error: get foo"));
     eng.evaluate("Object"); // clear exception state...

